pkg/services/frontend/index.html

```diff
@@ -239,6 +239,40 @@
 
         const CHECK_INTERVAL = 1 * 1000;
 
+        function getCookie(name) {
+          const cookies = document.cookie.split(";").map(c => c.trim());
+
+          for (const cookie of cookies) {
+            if (cookie.startsWith(name + "=")) {
+              return cookie.substring(name.length + 1);
+            }
+          }
+
+          return null;
+        }
+
+        function getSessionExpiration() {
+          const value = getCookie("grafana_session_expiry") || "0";
+          const realExpiresSeconds = parseInt(value, 10);
+          const expiresSeconds = Math.max(realExpiresSeconds - 10, 0); // Rotate 10s before the real expiration
+          const expiration = new Date(expiresSeconds * 1000);
+          return expiration;
+        }
+
+        async function rotateSession() {
+          const resp = await fetch('/api/user/auth-tokens/rotate', { method: 'POST' });
+
+          if (resp.status === 200) {
+            return true;
+          }
+
+          if (resp.status === 401) {
+            return false;
+          }
+
+          return undefined;
```

Comment:
does this have a purpose? 🤔 doesn't look like we do anything with the return value in `attemptFetch`